### ISSUE TITLE
In the source code, the use of attention_mask is contradictory

### ISSUE THREAD

**Issue by ghozn** [Ki6an/fastT5#66]
Hi,
I try to convert the onnx model to the tensorrt model, but it said that the input_ids and encoder_attention_mask must have the same sequence.
So I read your source code and find that encoder, init-decoder and decoder use the same attention_mask in **onnx_model.py** on line **168, 182 and 192**. But when you convert the torch model using code torch.onnx.export in **onnx_exporter.py** on line **177, 190 and 207**, you use attention_mask and attention_mask_dec. This is very confusing.

**Comment by ghozn** [comment]
`dyn_axis_general = {0: "batch", 1: "sequence"}`
`dyn_axis = {
            "input_ids": dyn_axis_general,
            "encoder_attention_mask": dyn_axis_general,
            "encoder_hidden_states": dyn_axis_general,
            "logits": dyn_axis_general,
            "output_past_key_values": dyn_axis_general,
        }`
The input_ids, encoder_attention_mask, encoder_hidden_states in the init-decoder should not have the same 1 dimension
